python/valis/routes/target.py

```diff
@@ -181,7 +182,18 @@ async def get_spectrum(self, sdss_id: Annotated[int, Path(title="The sdss_id of
                 response_model_exclude_unset=True, response_model_exclude_none=True)
     async def get_catalogs(self, sdss_id: int = Path(title="The sdss_id of the target to get", example=23326)):
         """ Return catalog information for a given sdss_id """
-        return get_catalog_sources(sdss_id).dicts().iterator()
+        sdss_id_data = get_catalog_sources(sdss_id).dicts()
+
+        parent_catalogs: list[dict[str, Any]] = []
+        for i in sdss_id_data:
+            cat_data = get_parent_catalogs(i['catalogid'])
+            if cat_data.count() > 0:
+                parent_catalogs.append(cat_data.dicts()[0])
+            else:
+                parent_catalogs.append({})
```

Comment:
why you are extracting and using the `catalogid` here for the query, rather than using the `sdss_id` itself?  `get_catalog_sources` returns all catalogids for a given sdssid, which I think is similar for the `sdss_id_to_catalog` table.  Would the two queries using `sdss_id` not necessarily return lists of matching elements?

Reply:
Another question, why not have the `parent_catalog` query and output be included in the output of`get_catalog_sources`?

Reply:
I think two main reasons:

1) I thought it may be useful to have a dedicated function (and maybe route?) to get the parent catalogue information.
2) More importantly, if I merge the `sdss_id_to_catalog` query in `get_catalog_sources` I'm not sure there's a good way to get the parent catalog IDs inside a dictionary instead of flattened with the other information (`ra`, `dec`, `epoch`, etc.). That can be fixed later of course, but it would require hardcoding some columns. Or we can just have the response include the parent catalogues without grouping them.

Reply:
A related question is whether we want to trim the returned string IDs. Some PKs are padded with spaces, for example

```console
>> curl -L "http://127.0.0.1:8000/target/catalogs/129055990"
[{"sdss_id":129055990,"ra_sdss_id":342.8413714603454,"dec_sdss_id":-10.239850309221548,"catalogid":63050395068163877,"n_associated":1,"ra_catalogid":342.8413714603454,"dec_catalogid":-10.239850309221548,"version":31,"lead":"gaia_dr3_source","ra":342.8413714603454,"dec":-10.239850309221548,"pmra":-21.31166,"pmdec":-41.54385,"parallax":10.907291,"parent_catalogs":{"allstar_dr17_synspec_rev1":"apogee.apo25m.stars.K2_C3_059-57.2M22512196-1014225","allwise":3430010701351038164,"gaia_dr2_source":2605275519066876160,"gaia_dr3_source":2605275489003187072,"legacy_survey_dr10":43981609979873293,"legacy_survey_dr8":8796110910785983,"panstarrs1":441101731249513,"sdss_dr13_photoobj":1237652598500819158,"sdss_dr17_specobj":"    815233483986200576","tic_v8":36598932,"twomass_psc":124799679}}]
```

As we serialise (or in a validator) we can trim `sdss_dr17_specobj` and the other strings, but maybe that's best done in the front-end.

Reply:
That makes sense.  Supposedly there's a way in peewee to control where you put the output from a join using the `attr` keyword, https://docs.peewee-orm.com/en/latest/peewee/relationships.html#selecting-from-multiple-sources, but I could never get this to work with to `.dicts()`.  I do like grouping them under `parent_catalogs`, so I think this is fine for now.  

Regarding the trimming, yes, I see what you mean.  I think we should strip them here in the serializer.   Are the `pks` stored in the original parent catalog with that extra whitespace?  Why is that pk a string but the rest in the example are integers?

Reply:
Ah, I seemed to remember that there was something like `attr` and was looking the documentation for it. Let me see if I can get that work.

Ok, I'll trim them. Yes, the way they show up in that curl above is how they are stored in the PK and in the `sdss_id_to_catalog` view. `sdss_dr17_specobj.specobj_id` is a string because it's larger than a 64-bit integer so it's a string that accepts up to a 128-bit integer (I think).

Reply:
Ok, that makes sense.  I still think we should trim them.  No user will care or remember about the padding.  Is `sdssdb` or peewee smart enough to deal with the whitespace if a user specifies `815233483986200576`?  or is this something we need to account for in the other route?

Reply:
No, if you do a query in Peewee (or PSQL) with the trimmed PK it won't match. We would need to re-pad it or use a `LIKE` (which would probably make the query slower but maybe not much).

Reply:
I think re-padding is ok if that column is always the same fixed-length.